### PR TITLE
[mono] Intrinsify multiple StoreVector API's

### PR DESCRIPTION
**Pull request by jkurdek**
Contributes to https://github.com/dotnet/runtime/issues/93081

This PR intrinsifies the following APIs:

AdvSimd.StoreVector64x*
AdvSimd.Arm64.StoreVector128x*
AdvSimd.StoreVector64x*AndZip
AdvSimd.Arm64.StoreVector128x*AndZip

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -4351,6 +4357,29 @@ emit_arm64_intrinsics (
...
+			case SN_StoreVector128x4: opcode = INTRINS_AARCH64_ADV_SIMD_ST1X4_V128; break;
+			case SN_StoreVector64x2: opcode = INTRINS_AARCH64_ADV_SIMD_ST1X2_V64; break;
+			case SN_StoreVector64x3: opcode = INTRINS_AARCH64_ADV_SIMD_ST1X3_V64; break;
+			case SN_StoreVector64x4: opcode = INTRINS_AARCH64_ADV_SIMD_ST1X4_V64; break;
+			default: g_assert_not_reached ();
+			}
+
+			MonoClass* value = mono_class_from_mono_type_internal (fsig->params [1]);
+
+			MonoInst* ins = emit_simd_ins_for_sig (cfg, klass, OP_ARM64_STM, opcode, arg0_type, fsig, args);
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:4379]: You could pass `value` as the second argument for `emit_simd_ins_for_sig` and get rid of the next line.

**Comment by kunalspathak** [comment]
cc: @a74nh @SwapnilGaikwad

**Review by kunalspathak — APPROVED** [review]
LGTM. Thanks!

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -4351,6 +4363,38 @@ emit_arm64_intrinsics (
...
+		case SN_StoreVector64x2:
+		case SN_StoreVector64x3:
+		case SN_StoreVector64x4:
+		case SN_StoreVector128x2AndZip:
+		case SN_StoreVector128x3AndZip:
+		case SN_StoreVector128x4AndZip:
+		case SN_StoreVector64x2AndZip:
+		case SN_StoreVector64x3AndZip:
+		case SN_StoreVector64x4AndZip: {
+			int opcode = 0;
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:4378]: Usually we consider `OP_ARM64_STM` as the opcode. I would prefer to rename this variable as `intrinsics_id` or `iid`

**Review by fanyang-mono — APPROVED** [review]
LGTM!

**Pull request merged**